polygeist/lib/Dialect/Polygeist/Transforms/SYCLRaiseHost.cpp

```diff
@@ -49,15 +50,94 @@ class SYCLRaiseHostConstructsPass
 // Pattern
 //===----------------------------------------------------------------------===//
 
+namespace {
+struct RaiseKernelName : public OpRewritePattern<LLVM::GlobalOp> {
+public:
+  using OpRewritePattern<LLVM::GlobalOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(LLVM::GlobalOp op,
+                                PatternRewriter &rewriter) const final {
+    // Get a reference to the kernel this global references
+    std::optional<SymbolRefAttr> ref = getKernelRef(op);
+    if (!ref)
+      return failure();
+
+    rewriter.replaceOpWithNewOp<sycl::SYCLHostKernelNameOp>(op, op.getSymName(),
```

Comment:
If there are multiple users of the global and we replace it the remaining users would reference an invalid operation.